### PR TITLE
Fix aes cfb1 so that it can operate in bit mode.

### PR DESCRIPTION
**Pull request by slontis**
The code to handle the cipher operation was already in the provider.
It just needed a OSSL_PARAM in order to set this into the algorithm.
EVP_CIPHER_CTX_set_flags() has been modified to pass the OSSL_PARAM.

Issue reported by Mark Powers from Acumen.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Made this a draft because documentation is still needed for the new parameter, but this should be only done after the cipher documentation PR has been merged.

**Comment by slontis** [comment]
NOTE: There is now a bit of a problem using the EVP_CIPHER_CTX_set_flags() in that it has to be done AFTER the cipher has been attached to the cipher_ctx. Before it just set ctx->flags but now it needs access to ctx->cipher->set_ctx_params()..

**Review by paulidale — APPROVED** [review]
LGTM.

Perhaps raise an issue for the documentation to be completed before final.  This should really be in for the beta.

**Comment by paulidale** [comment]
It depends on the documentation.  If it's going to be part of this, it isn't ready to merge.  If it will be a separate PR, this is ready.

**Comment by slontis** [comment]
Updated documentation now that the "Update cipher documentation" PR has been merged. Changes are in the last 2 commits.

**Comment by slontis** [comment]
ping

File: include/openssl/core_names.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ extern "C" {
 
 /* cipher parameters */
 #define OSSL_CIPHER_PARAM_PADDING              "padding"      /* uint */
+#define OSSL_CIPHER_PARAM_USE_BITS             "use-bits"     /* uint */
```
Comment (paulidale) [inline include/openssl/core_names.h:68]: Is a more descriptive name possible here?  `"bit-mode"`?
Just the string, the macro name is okay.

**Review by paulidale — APPROVED** [review]

File: include/openssl/core_names.h (hunk 1 above)
Comment (t8m) [reply include/openssl/core_names.h:68]: Or `length-in-bits`? But then it is still unclear which length.

Comment (t8m) [reply include/openssl/core_names.h:68]: Or `length-is-bits`

Comment (paulidale) [reply include/openssl/core_names.h:68]: `bit-at-a-time`

Comment (slontis) [reply include/openssl/core_names.h:68]: All those names seem to suck :)

Comment (paulidale) [reply include/openssl/core_names.h:68]: More or less than the incumbent?

Comment (slontis) [reply include/openssl/core_names.h:68]: about the same..

**Comment by slontis** [comment]
Merged to master.
Feel free to add another PR that changes the name if required.

**Pull request closed without merging**